engine/src/main/java/org/hibernate/search/backend/impl/lucene/works/AddWorkExecutor.java

```diff
@@ -84,7 +85,7 @@ static ScopedAnalyzerReference updateAnalyzerMappings(Workspace workspace, Scope
 			return scopedAnalyzerReference;
 		}
 
-		ScopedAnalyzerReference.Builder copyBuilder = new ScopedAnalyzerReference.Builder( scopedAnalyzerReference );
+		ScopedAnalyzer.Builder copyBuilder = scopedAnalyzerReference.getAnalyzer().startCopy();
```

Comment:
It's weird. It looks like we are copying an analyzer while in fact we build an AnalyzerReference. Maybe the name `Builder` is misleading and `ReferenceBuilder` would be better?

All in all, I'm not convinced this is an improvement. What's the exact reason for this change (the commit message does not really help).